### PR TITLE
[23.2] Fix succces typo

### PR DESCRIPTION
**Pull request by mvdbeek**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/model/__init__.py [hunk 1]
```diff
@@ -4409,13 +4409,13 @@ def state(self, state: Optional[DatasetState]):
...
             else:
-                self.set_metadata_succces_state()
+                self.set_metadata_succcess_state()
                 sa_session = object_session(self)
                 if sa_session:
                     sa_session.add(self.dataset)
                 self.dataset.state = state
 
-    def set_metadata_succces_state(self):
+    def set_metadata_succcess_state(self):
```
Comment (nsoranzo) [inline lib/galaxy/model/__init__.py]:
```suggestion
    def set_metadata_success_state(self):
```

Comment (mvdbeek) [reply lib/galaxy/model/__init__.py]: lol, I don't know what's up with my spellchecker ... 😅

**Review by nsoranzo — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/controllers/dataset.py [hunk 2]
```diff
@@ -361,8 +361,8 @@ def set_edit(self, trans, payload=None, **kwd):
                     annotation = sanitize_html(payload.get("annotation"))
                     self.add_item_annotation(trans.sa_session, trans.get_user(), data, annotation)
                 # if setting metadata previously failed and all required elements have now been set, clear the failed state.
-                if data._state == trans.model.Dataset.states.FAILED_METADATA and not data.missing_meta():
-                    data._state = None
+                if data.state == trans.model.Dataset.states.FAILED_METADATA and not data.missing_meta():
```
Comment (martenson) [inline lib/galaxy/webapps/galaxy/controllers/dataset.py:364]: Given the PR title just making sure you want this here.

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/controllers/dataset.py:364]: Look at that, I guess search and replace led me to save a pending change ... I didn't mean to, but this is the right thing to do.

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**